include/exiv2/jp2image.hpp

```diff
@@ -42,7 +42,7 @@ namespace Exiv2
     // Add JPEG-2000 to the supported image formats
     namespace ImageType
     {
-        const int jp2 = 15;                     //!< JPEG-2000 image type
+        const int jp2 = 19;                     //!< JPEG-2000 image type
```

Comment:
Why is the new type using the value 15 and `jp2` value has been updated from 15 to 19? Was not possible to keep the previous values for jp2 and use a new one for BMFF?

Reply:
15 is used by bmff, and 19 by jp2.

Let's replace that with an enum in 'main'.  I didn't want change it to an enum for a "dot" because _if it's not broken, don't fix it_.  However, when we get to 'main/0.28' we can fix this.

```bash
509 rmills@rmillsm1:~/gnu/github/exiv2/bmff $ grep 'const int' include/exiv2/*image.hpp 
include/exiv2/bmffimage.hpp:        const int bmff = 15;  //!< BMFF (bmff) image type (see class BMFF)
include/exiv2/bmpimage.hpp:        const int bmp = 14; //!< Windows bitmap (bmp) image type (see class BmpImage)
include/exiv2/cr2image.hpp:        const int cr2 = 7;          //!< CR2 image type (see class Cr2Image)
include/exiv2/crwimage.hpp:        const int crw = 3;          //!< CRW image type (see class CrwImage)
include/exiv2/epsimage.hpp:        const int eps = 18;                     //!< EPS image type
include/exiv2/gifimage.hpp:        const int gif = 11;          //!< GIF image type (see class GifImage)
include/exiv2/image.hpp:        const int none = 0;         //!< Not an image
include/exiv2/jp2image.hpp:        const int jp2 = 19;                     //!< JPEG-2000 image type
include/exiv2/jpgimage.hpp:        const int jpeg = 1;         //!< JPEG image type (see class JpegImage)
include/exiv2/jpgimage.hpp:        const int exv  = 2;         //!< EXV image type (see class ExvImage)
include/exiv2/mrwimage.hpp:        const int mrw = 5;          //!< MRW image type (see class MrwImage)
include/exiv2/orfimage.hpp:        const int orf = 9;          //!< ORF image type (see class OrfImage)
include/exiv2/pgfimage.hpp:        const int pgf = 17;          //!< PGF image type (see class PgfImage)
include/exiv2/pngimage.hpp:        const int png = 6;          //!< PNG image type (see class PngImage)
include/exiv2/psdimage.hpp:        const int psd = 12; //!< Photoshop (PSD) image type (see class PsdImage)
include/exiv2/rafimage.hpp:        const int raf = 8;          //!< RAF image type (see class RafImage)
include/exiv2/rw2image.hpp:        const int rw2 = 16;             //!< RW2 image type (see class Rw2Image)
include/exiv2/tgaimage.hpp:        const int tga = 13; //!< Truevision TARGA (tga) image type (see class TgaImage)
include/exiv2/tiffimage.hpp:        const int tiff = 4;          //!< TIFF image type (see class TiffImage)
include/exiv2/tiffimage.hpp:        const int dng = 4;           //!< DNG image type (see class TiffImage)
include/exiv2/tiffimage.hpp:        const int nef = 4;           //!< NEF image type (see class TiffImage)
include/exiv2/tiffimage.hpp:        const int pef = 4;           //!< PEF image type (see class TiffImage)
include/exiv2/tiffimage.hpp:        const int arw = 4;           //!< ARW image type (see class TiffImage)
include/exiv2/tiffimage.hpp:        const int sr2 = 4;           //!< SR2 image type (see class TiffImage)
include/exiv2/tiffimage.hpp:        const int srw = 4;           //!< SRW image type (see class TiffImage)
include/exiv2/webpimage.hpp:        const int webp = 23; //!< Treating webp as an image type>
510 rmills@rmillsm1:~/gnu/github/exiv2/bmff $ 
```

Reply:
I've realised during lunch that you're asking "why not leave jp2 unchanged and set bmff=19?".   When I started working with Peter, the only way I could change anything was to send him a patch and that's what I did.  I didn't want to discuss what that code means, so I accepted his 15 (even though it's wrong).

You're right.  Now that we're cleaning up and the end, I should restore jp2 = 15 and set bmff = 19.  _**Good Catch.**_